### DISCUSSION TITLE
More error details in API response

### DISCUSSION THREAD

**Discussion by basavarajdodamani** [strapi/strapi#7831] [Legacy Topics]
Hello,

Could we have more error details sent by STRAPI api? As an example, console captures if duplicate value is getting inserted in an unique field of a collection. But in the API response only below details are sent:
{
    "statusCode": 500,
    "error": "Internal Server Error",
    "message": "An internal server error occurred"
}

Could we have below error stack in the response:
![image](https://user-images.githubusercontent.com/69140968/92732166-b293e380-f393-11ea-9b47-788ed2da4735.png)

Your response is highly appreciated.

Kind Regards,
Basu

**Comment by derrickmehaffy** [comment]
It is not a good practice to share additional information in the case of a server error to the client, this provides an avenue of attack to give attackers more information.

A 500 error is always a server error, whereas a 400 error is one the client caused, hence why more information is provided to them to fix the problem on their side.

**Reply by basavarajdodamani** [reply]
> 
> 
> It is not a good practice to share additional information in the case of a server error to the client, this provides an avenue of attack to give attackers more information.
> 
> A 500 error is always a server error, whereas a 400 error is one the client caused, hence why more information is provided to them to fix the problem on their side.

Thanks @derrickmehaffy  for the quick response. Noted. My intention is to capture this detailed message for troubleshooting purpose. Any suggestions to load this console message into a file?

**Reply by derrickmehaffy** [reply]
I would suggest you use something like [Sentry](https://sentry.io/welcome/) for this, as we also have a guide on doing so here: https://strapi.io/documentation/v3.x/guides/error-catching.html#create-a-middleware

**Reply by basavarajdodamani** [reply]
okay I shall try this out.. thank you..

**Reply by derrickmehaffy** [reply]
(Sentry is free when self-hosted, I use it myself) And you are welcome to take a look at my own personal middleware: https://github.com/canonn-science/CAPIv2-Strapi/blob/development/middlewares/sentry/index.js
